File: 3355-zero-array-transformation-i/3355-zero-array-transformation-i.java
class Solution {
    public boolean isZeroArray(int[] nums, int[][] queries) {
        int n = nums.length;
        int[] check = new int[n + 1];
        
        for (int[] x : queries) {
            int l = x[0];
            int r = x[1];
            check[l] -= 1;
            if (r + 1 < n) {
                check[r + 1] += 1;
            }
        }
        
        int temp = 0;
        for (int i = 0; i < n; i++) {
            temp += check[i];
            nums[i] += temp;
            if (nums[i] < 0) nums[i] = 0;
        }
        
        for (int i = 0; i < n; i++) {
            if (nums[i] != 0) return false;
        }
        return true;
    }
}

File: 3355-zero-array-transformation-i/3355-zero-array-transformation-i.py
class Solution:
    def isZeroArray(self, nums: List[int], queries: List[List[int]]) -> bool:
        n = len(nums)
        check = [0] * (n + 1)
        
        for x in queries:
            l = x[0]
            r = x[1]
            check[l] -= 1
            if r + 1 < n:
                check[r + 1] += 1
        
        temp = 0
        for i in range(n):
            temp += check[i]
            nums[i] += temp
            if nums[i] < 0:
                nums[i] = 0
        
        return all(x == 0 for x in nums)